curp/src/client.rs

```diff
@@ -402,7 +402,10 @@ where
                 .write()
                 .check_and_update(resp.leader_id, resp.term);
             match resp.error {
-                Some(e) => return Err(bincode::deserialize(&e)?),
+                Some(e) => {
+                    warn!("shutdown error: {:?}", e);
+                    continue;
```

Comment:
I think we should add a comment explaining why this error can be retried instead of returning it to the user.

Reply:
Added